.github/workflows/digest-updater.yaml

```diff
@@ -0,0 +1,87 @@
+# The format of the digest that would be assing has the following format: quay.io/opendatahub/workbench-images@sha256:xxx...xxx
+on:
+  workflow_dispatch:
+    inputs:
+      branch:
+        required: true
+        description: "Provide release branch"
+jobs:
+  initialize:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+      - name: Install Skopeo CLI
+        shell: bash
+        run: |
+          sudo apt-get -y update
+          sudo apt-get -y install skopeo
+  update-n-version:
+    needs: [ initialize ]
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    strategy:
+      max-parallel: 1
+      matrix:
+        files: [ "notebook-images/base/jupyter-minimal-notebook-imagestream.yaml", "notebook-images/base/jupyter-datascience-notebook-imagestream.yaml", "notebook-images/base/jupyter-minimal-gpu-notebook-imagestream.yaml" , "notebook-images/base/jupyter-pytorch-notebook-imagestream.yaml","notebook-images/base/jupyter-tensorflow-notebook-imagestream.yaml" ]
+
+    steps:
+      - name: Configure Git
+        run: |
+         git config --global user.email "github-actions[bot]@users.noreply.github.com"
+         git config --global user.name "GitHub Actions"
+
+      # Get the latest weekly build commit hash: https://github.com/opendatahub-io/notebooks/commits/2023a
+      - name: Checkout upstream notebooks repo
+        uses: actions/checkout@v3
+        with:
+         repository: opendatahub-io/notebooks.git
+         ref: 2023a
+      - name: Retrive latest weekly commit hash from the release branch
+        id: hash
+        shell: bash
+        run: |
+          echo "HASH=$(git rev-parse --short HEAD)" >> ${GITHUB_OUTPUT}
+
+      - name: Checkout main from the odh-manifest repo
+        uses: actions/checkout@v3
+        with:
+          ref: main
+      # Get image name from: .spec.tags[0].from.name
+      - name: Fetch the name of the image from Image Stream (Step 1)
+        shell: bash
+        id: img_name
+        run: |
+         echo "img_name=$(yq '.spec.tags[0].from.name' ${{ matrix.files }} | sed 's/\"/\//g')" >> ${GITHUB_OUTPUT}
+
+      # Checkout the release branch to apply the updates
+      - name: Checkout release branch
+        uses: actions/checkout@v3
+        with:
+          ref:  ${{ inputs.branch }}
+
+      - name: Fetch digest, and update the imagestream file (Step 2)
+        env:
+          IMG_NAME: ${{ steps.img_name.outputs.img_name }}
+        run: |
+          src_tag=$(echo "${{ env.IMG_NAME }}" | sed 's/-weekly.*//' | cut -d':' -f2 | tr -d '"')
+          regex="$src_tag-\d+-${{ steps.hash.outputs.HASH }}"
+          echo $regex
+
+          # Get the latest tag: output ex. latest_tag=jupyter-minimal-ubi9-python-3.9-2023a-20230403-cf92364
```

Comment:
If we are using `*-weekly` tags for the imagestreams in odh-manifests and with the expectation that it is a floating tag that will be updated weekly, can we simplify this to  use the `skopeo inspect <IMAGE:TAG> | jq -r .Digest` as the latest digest hash?

Reply:
The weekly isn't a floating tag, its lifetime is only for one week then replaces by the new rebuild, thus the digest doesn't remain stable over the time. The `notebook-release-YYYYMMDD-git-hash` on the other side remains the same always.

Reply:
in my view, floating tag means exactly that: the the tag itself is fixed, e.g. blabla-weekly, but the digest behind the image changes regularly. See https://redhat-connect.gitbook.io/catalog-help/container-images/container-image-details/image-tags-and-versions. Very good stat release-type images stay unchanged and do not chnage their sha digest value.